--- a/drivers/usb/serial/usb_wwan.c
+++ b/drivers/usb/serial/usb_wwan.c
@@ -471,5 +471,18 @@ static struct urb *usb_wwan_setup_urb(struct usb_serial_port *port,
 			  usb_sndbulkpipe(serial->dev, endpoint) | dir,
 			  buf, len, callback, ctx);

+//Added by Quectel for zero packet
+	if (dir == USB_DIR_OUT) {
+	  struct usb_device_descriptor *desc = &serial->dev->descriptor;
+	  if (desc->idVendor == cpu_to_le16(0x05C6) && desc->idProduct == cpu_to_le16(0x9090))
+	    urb->transfer_flags |= URB_ZERO_PACKET;
+	  if (desc->idVendor == cpu_to_le16(0x05C6) && desc->idProduct == cpu_to_le16(0x9003))
+	    urb->transfer_flags |= URB_ZERO_PACKET;
+	  if (desc->idVendor == cpu_to_le16(0x05C6) && desc->idProduct == cpu_to_le16(0x9215))
+	    urb->transfer_flags |= URB_ZERO_PACKET;
+	  if (desc->idVendor == cpu_to_le16(0x2C7C))
+	    urb->transfer_flags |= URB_ZERO_PACKET;
+	}
+
 	return urb;
 }
